pkg/controller/endpointslice/endpointslice_controller.go

```diff
@@ -313,6 +313,12 @@ func (c *Controller) syncService(key string) error {
 		return err
 	}
 
+	if service.DeletionTimestamp != nil {
+		// If a Service is being deleted, return immediately and leave any
+		// related EndpointSlices to be cleaned up by garbage collector.
+		return nil
```

Comment:
should anything with triggerTimeTracker and reconciler be done here similar to the NotFound case?

Reply:
Updated, thanks!